distribution.pm

```diff
@@ -238,8 +239,9 @@ sub script_output ($self, $script, @args) {
             type_command => length($script) < 80,
         }, ['timeout'], @args);
 
-    my $marker = testapi::hashed_string("SO$script");
+    my $marker = testapi::hashed_string("SO$script") . $count;
     my $script_path = "/tmp/script$marker.sh";
+    $count++;
```

Comment:
```suggestion
    my $marker = testapi::hashed_string("SO$script") . $count++;
    my $script_path = "/tmp/script$marker.sh";
```

You might also want to move `state $count = 0;` directly before `my $marker …` so everything that belongs together is actually together.

And if you have everything together you can also split the code easily into a separate function, e.g. `make_script_output_marker`. Then you can also test this function separately in unit tests.